zap/src/test/java/org/zaproxy/zap/utils/ZapXmlConfigurationUnitTest.java

```diff
@@ -43,6 +49,87 @@ class ZapXmlConfigurationUnitTest {
                     + "    </d>\n"
                     + "</a>\n";
 
+    @Test
+    void shouldBeInstantiatedWithDelimterParsingDisabled() {
+        // Given
+        ZapXmlConfiguration config = new ZapXmlConfiguration();
+        // When
+        boolean isParsingDisabled = config.isDelimiterParsingDisabled();
+        // Then
+        assertThat(isParsingDisabled, is(equalTo(true)));
+    }
+
+    @Test
+    void shouldSetVariousPropertiesAsExpected() throws ConfigurationException, UnsupportedEncodingException {
+        // Given
+        ZapXmlConfiguration config = new ZapXmlConfiguration();
+        String aString = "Some string 🙂.";
+        String bString = "";
+        String cString = null;
+        int aInt = 120;
+        boolean aBoolean = true;
+        // When
+        config.setProperty("aString", aString);
+        config.setProperty("bString", bString);
+        config.setProperty("cString", cString);
+        config.setProperty("aInt", aInt);
+        config.setProperty("aBoolean", aBoolean);
+        List<Object> aStringList = config.getList("aString");
+        List<Object> bStringList = config.getList("bString");
+        List<Object> cStringList = config.getList("cString");
+        List<Object> aIntList = config.getList("aInt");
+        int aActualInt = config.getInt("aInt");
+        List<Object> aBooleanList = config.getList("aBoolean");
+        boolean aActualBoolean = config.getBoolean("aBoolean");
+        // Then
+        assertThat(aStringList.size(), is(equalTo(1)));
+        assertThat(aStringList.get(0), is(equalTo(aString)));
+        assertThat(bStringList.size(), is(equalTo(1)));
+        assertThat(bStringList.get(0), is(equalTo(bString)));
+        assertThat(cStringList.size(), is(equalTo(0)));
+        assertThat(aIntList.size(), is(equalTo(1)));
+        // Int is string when handled via list
+        assertThat(aIntList.get(0), is(equalTo("120")));
+        assertThat(aActualInt, is(equalTo(120)));
+        assertThat(aBooleanList.size(), is(equalTo(1)));
+        // Boolean is string when handled via list
+        assertThat(aBooleanList.get(0), is(equalTo("true")));
+        assertThat(aActualBoolean, is(equalTo(true)));
+    }
+
+    @Test
+    @Disabled("Should not split on comma, when setting list")
```

Comment:
Including `super.setListDelimiter('\0')` in the constructor does make this test pass.